### PR TITLE
SK0 37

### PR DESCRIPTION
**Pull request by rolcho**
- profile page created
- fix(be): UsersService test
- updated test
- fix(fe): profile page
- fix(be) UsersService and test

### CODE REVIEW

File: sk0-back/src/users/controllers/users.controller.ts [hunk 1]
```diff
@@ -62,6 +62,8 @@ export class UsersController {
     try {
       return this.authService.createUser(createUserDto);
     } catch (err) {
+      console.log("hello");
```
Comment (markkovari) [inline sk0-back/src/users/controllers/users.controller.ts]: Should be removed.

File: sk0-front/jest.config.js [hunk 2]
```diff
@@ -1,5 +1,9 @@
 module.exports = {
   preset: "jest-preset-angular",
-  testPathIgnorePatterns: ["<rootDir>/dist/", "<rootDir>/node_modules/"],
+  testPathIgnorePatterns: [
+    "<rootDir>/dist/",
+    "<rootDir>/node_modules/",
+    "<rootDir>/.angular/"
+  ],
```
Comment (markkovari) [inline sk0-front/jest.config.js:7]: 🙏

File: sk0-front/src/app/profile/profile.page.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    IonicModule,
+    CommonModule,
+    FormsModule,
+    ReactiveFormsModule,
+    RouterModule
+  ]
+})
+export class ProfilePage {
+  private readonly strongPasswordValidator =
+    /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{6,}$/;
```
Comment (markkovari) [inline sk0-front/src/app/profile/profile.page.ts:33]: Later we might want to move the reused part into a common service or util. But looks good for now.

File: sk0-front/src/app/profile/profile.page.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    private http: HttpClient,
+    private navCtl: NavController
+  ) {
+    this.http
+      .get<LoginResponse>(`${environment.baseUrl}/users/${this.userId$.value}`)
+      .subscribe({
+        next: async (response) => {
+          this.profileForm.get("name")?.setValue(response.name),
+            this.profileForm.get("email")?.setValue(response.email);
+        }
```
Comment (markkovari) [inline sk0-front/src/app/profile/profile.page.ts]:
```suggestion
          this.profileForm.get("name")?.setValue(response.name);
          this.profileForm.get("email")?.setValue(response.email);
        }
```

File: sk0-front/src/app/profile/profile.page.html [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                ></ion-input>
+              </ion-item>
+              <ion-button
+                class="ion-margin-top"
+                (click)="onUpdate()"
+                color="primary"
+                expand="block"
+                >Módosítás</ion-button
+              >
+
```
Comment (markkovari) [inline sk0-front/src/app/profile/profile.page.html]:
```suggestion
```

**Review by markkovari — CHANGES_REQUESTED** [review]
Good job. We should not really have bigger PRs. This is a good context boundary.

File: sk0-front/src/app/profile/profile.page.ts (hunk 3 above)
Comment (rolcho) [reply sk0-front/src/app/profile/profile.page.ts:33]: @markkovari Yes we should talk about how and where to store

Comment (markkovari) [reply sk0-front/src/app/profile/profile.page.ts:33]: Yesyes, closing this convo. :pray:

File: sk0-back/src/users/controllers/users.controller.ts (hunk 1 above)
Comment (rolcho) [reply sk0-back/src/users/controllers/users.controller.ts]: sure

**Review by markkovari — APPROVED** [review]

**Review by bendesbeus — APPROVED** [review]

**Review by Jankita58 — APPROVED** [review]

**Review by markkovari — APPROVED** [review]

**Pull request merged**